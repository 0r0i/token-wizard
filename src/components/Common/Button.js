import React from 'react'

export const Button = ({ buttonText = '', disabled = false, extraClassName = '', onClick, type = 'button' }) => (
  <button className={`sw-Button ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    {buttonText}
  </button>
)
