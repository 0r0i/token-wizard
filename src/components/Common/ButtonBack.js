import React, { Component } from 'react'

export class ButtonBack extends Component {
  /**
   * Render method for ButtonBack component
   * @returns {*}
   */
  render() {
    const { onClick } = this.props

    return (
      <button onClick={onClick} type="button" className={`sw-ButtonBack`}>
        Back
      </button>
    )
  }
}
