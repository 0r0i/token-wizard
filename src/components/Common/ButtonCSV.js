import React from 'react'

export const ButtonCSV = ({ onClick, text, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonCSV ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    {text}
  </button>
)
