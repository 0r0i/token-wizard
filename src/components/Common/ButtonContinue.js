import React from 'react'

export const ButtonContinue = ({ disabled = false, type, onClick, extraClassName, buttonText = 'Continue' }) => (
  <button
    className={`sw-ButtonContinue ${extraClassName ? extraClassName : ''}`}
    disabled={disabled}
    onClick={onClick}
    type={type}
  >
    <span className={`sw-ButtonContinue_Text`}>{buttonText}</span>
  </button>
)
