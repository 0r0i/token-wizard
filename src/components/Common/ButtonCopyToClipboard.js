import React from 'react'

export const ButtonCopyToClipboard = ({ disabled, extraClassName, value }) => (
  <button
    className={`sw-ButtonCopyToClipboard ${extraClassName ? extraClassName : ''}`}
    data-clipboard-action="copy"
    data-clipboard-text={value}
    disabled={disabled}
    type="button"
  />
)
