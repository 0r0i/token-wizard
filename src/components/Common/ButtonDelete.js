import React from 'react'

export const ButtonDelete = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonDelete ${extraClassName}`} disabled={disabled} onClick={onClick} type="button" />
)
