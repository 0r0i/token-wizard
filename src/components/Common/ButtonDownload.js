import React, { Component } from 'react'

export class ButtonDownload extends Component {
  render() {
    const { disabled, onClick, extraClass } = this.props
    return (
      <button
        className={`sw-ButtonDownload ${extraClass ? extraClass : ''}`}
        disabled={disabled}
        onClick={onClick}
        type="button"
      >
        Download
      </button>
    )
  }
}
