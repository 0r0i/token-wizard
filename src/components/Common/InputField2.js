import React from 'react'
import { Errors } from './Errors'
import { FormControlTitle } from '../Common/FormControlTitle'
import { TextField } from '../Common/TextField'

export const InputField2 = props => {
  if (props.val) {
    props.input.value = props.val
  }

  return (
    <div className={`sw-InputField2 ${props.extraClassName ? props.extraClassName : ''}`}>
      {props.label ? <FormControlTitle title={props.label} description={props.description} /> : null}
      <TextField
        autoComplete="off"
        className={props.inputClassName ? props.inputClassName : 'input'}
        disabled={props.disabled}
        id={props.input.name}
        placeholder={props.placeholder}
        type={props.type}
        min={props.min}
        max={props.max}
        {...props.input}
      />
      <Errors name={props.input.name} />
    </div>
  )
}
