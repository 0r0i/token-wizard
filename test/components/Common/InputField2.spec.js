import React from 'react'
import {InputField2} from '../../../src/components/Common/InputField2'
import { Form } from 'react-final-form'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount, shallow } from 'enzyme'

configure({ adapter: new Adapter() })

describe('InputField2', () => {
  it(`should render InputField2 component`, () => {
    const wrapper = shallow(
      <InputField2
        input={{
          name: 'name',
          type: 'text'
        }}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it(`should render InputField2 component with given name`, () => {
    let input = {
      name: 'CustomName'
    }
    const wrapper = mount(
      <Form
        onSubmit={jest.fn()}
        component={InputField2}
        errorStyle={{ color: 'red', fontWeight: 'bold'}}
        input={input}
      />
    )
    expect(wrapper.find('label').props().htmlFor).toBe(input.name)
    expect(wrapper.find('input').props().id).toBe(input.name)
    expect(wrapper.find('Error').props().name).toBe(input.name)
  })

  it(`should be able to set value `, () => {
    const val = '22'
    let input = {
      name: 'CustomName',
      defaultValue: val,
      type: 'text'
    }
    const wrapper = mount(
      <Form
        onSubmit={jest.fn()}
        component={InputField2}
        errorStyle={{ color: 'red', fontWeight: 'bold'}}
        input={input}
      />
    )
    wrapper.find('input').simulate('change', { target: { defaultValue: val }})
    expect(wrapper.find('input').props().defaultValue).toBe(val)
  })
})
