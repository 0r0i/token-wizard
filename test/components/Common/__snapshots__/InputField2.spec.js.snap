// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField2 should render InputField2 component 1`] = `
<div>
  <label
    className="label"
    htmlFor="name"
  />
  <input
    autoComplete="off"
    className="input"
    id="name"
    name="name"
    type="text"
  />
  <p
    className="description"
  />
  <Error
    name="name"
  />
</div>
`;
